pkg/workflow/providers/oam/apply.go

```diff
@@ -146,10 +206,82 @@ func lookUpValues(v *value.Value) (*common.ApplicationComponent, *value.Value, s
 	return comp, patcher, clusterName, overrideNamespace, env, nil
 }
 
+func (p *provider) loadDynamicComponent(comp *common.ApplicationComponent) (*common.ApplicationComponent, error) {
+	if comp.Type == "ref-objects" {
```

Comment:
generally, we return the smaller logic first to avoid complex, so here we recommand:

```
if comp.Type!="ref-objects"{
  return ...
}

_comp:==comp.DeepCopy()
...
```

Reply:
OK